### PR TITLE
fix: move job name length to user settings 

### PR DESCRIPTION
**Pull request by banusha2**
## Context

Moving User's own preferred Job name length option,User Preference to User Settings .

## Objective
To provide more options in User Settings.

## References
Screen Shot 2022-08-31 at 2 38 36 PM

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/user-setting/component.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  maxDisplayLength: MAXIMUM_JOBNAME_LENGTH,
+
+  // Updating settings
+  async init() {
+    this._super(...arguments);
+
+    let desiredJobNameLength = MINIMUM_JOBNAME_LENGTH;
+
+    const pipelinePreference = await this.shuttle.getUserPipelinePreference(
+      this.get('pipeline.id')
```
Comment (adong) [inline app/components/user-setting/component.js:23]: I would think `pipeline.id` is undedfined after moving to global settings.

So, need to update the shuttle service (API service) to take care of the API endpoint changes _(be mindful about the backward copatablity, not to break the existing ones.)_ 

See: PR https://github.com/screwdriver-cd/data-schema/pull/503

File: app/components/user-setting/component.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    if (pipelinePreference) {
+      desiredJobNameLength = pipelinePreference.displayJobNameLength;
+    }
+
+    this.setProperties({ desiredJobNameLength });
+  },
+
+  async updateJobNameLength(displayJobNameLength) {
+    const pipelineId = this.get('pipeline.id');
```
Comment (adong) [inline app/components/user-setting/component.js:34]: Similarly

File: app/components/user-setting/component.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    const pipelinePreference = await this.shuttle.getUserPipelinePreference(
+      pipelineId
+    );
+
+    set(pipelinePreference, 'displayJobNameLength', displayJobNameLength);
+    pipelinePreference
+      .save()
+      .then(() =>
+        this.shuttle.updateUserPreference(pipelineId, pipelinePreference)
```
Comment (adong) [inline app/components/user-setting/component.js:43]: Here as well

File: app/components/user-setting/component.js (hunk 1 above)
Comment (jithine) [reply app/components/user-setting/component.js:23]: why pipeline is involved ?

Comment (banusha2) [reply app/components/user-setting/component.js]: Fixed it now, earlier it was as per old data-schema

File: app/components/user-setting/component.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  },
+
+  async updateUserSettings() {
+    this.set('isSaving', true);
+    const userSetting = await this.shuttle.getUserSetting();
+
+    try {
+      set(userSetting, 'displayJobNameLength', this.displayJobNameLength);
+      await this.shuttle.updateUserSettings(userSetting);
+      this.set('isSaving', false);
```
Comment (adong) [inline app/components/user-setting/component.js:39]: Minor
```
    this.setProperties({ });
```

File: app/components/user-setting/component.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  actions: {
+    async updateUserSettings() {
+      this.updateUserSettings();
+    },
+    async resetUserSettings() {
+      this.set('isSaving', true);
+      try {
+        await this.shuttle.deleteUserSettings();
+        this.set('isSaving', false);
+        this.set('successMessage', 'Timestamp preference reseted successfully');
```
Comment (adong) [inline app/components/user-setting/component.js:54]: Minor
```
    this.setProperties({ });
```

File: app/components/user-setting/component.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      } catch {
+        this.set('isSaving', false);
+        this.set(
```
Comment (adong) [inline app/components/user-setting/component.js:57]: Minor
```
    this.setProperties({ });
```

**Pull request closed without merging**